presentational/RepeatView.js

```diff
@@ -0,0 +1,218 @@
+  border: none;
+  color: ${({ theme }) => theme.colors.black};
+  font-family: ${({ theme }) => theme.fonts.text};
+  font-size: 15px;
+  font-weight: 300;
+  margin-left: 20px;
+`
+const Sort = styled.button`
+  background-color: ${({ theme }) => theme.colors.white};
+  border: none;
+  color: ${({ theme }) => theme.colors.black};
+  font-family: ${({ theme }) => theme.fonts.text};
+  font-size: 15px;
+  font-weight: 300;
+  margin-left: 20px;
+`
+const Tags = styled.button`
+  background-color: ${({ theme }) => theme.colors.white};
+  border: none;
+  color: ${({ theme }) => theme.colors.black};
+  font-family: ${({ theme }) => theme.fonts.text};
+  font-size: 15px;
+  font-weight: 300;
+  margin-left: 20px;
+`
+
+const DeckWrapper = styled.div`
+  width: auto;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  align-content: flex-start;
+  background-color: ${({ theme }) => theme.colors.white};
+`
+
+const DeckBox = styled.div`
+  display: flex;
+  flex: 0 0 auto;
+  width: 150px;
+  height: 150px;
+  padding: 5px;
+  margin: 10px;
+  border-radius: 10px;
+  justify-content: center;
+  align-items: center;
+  background-color: ${({ theme }) => theme.colors.primaryMuted};
+  font-family: ${({ theme }) => theme.fonts.text};
+  font-size: 30px;
+  font-weight: 600;
+  color: ${({ theme }) => theme.fonts.white};
+`
+const StyledLink = styled.a`
+  text-decoration: none;
+  opacity: ${({ active, theme }) =>
+    active ? 1 : theme.transparencies.inactive};
+
+  :hover {
+    opacity: 1;
+  }
+`
+
+export default ({
+  deckNames,
+  value,
+  changeFunction,
+  submitFunction,
+  modalState,
+  dismissModal
+}) => (
+  <RepeatWrapper>
+    <ViewBar>
+      <Browse>Browse</Browse>
+      <Decks>My Decks</Decks>
+    </ViewBar>
+    <DeckView>
+      <SearchWrapper>
+        <SearchBar>
+          <SearchBox placeholder="Start typing to search" />
+          <SearchButton>
+            <StyledSearchIcon />
+          </SearchButton>
+        </SearchBar>
+        <Filter>Filter</Filter>
+        <Sort>Sort</Sort>
+        <Tags>Tags</Tags>
+      </SearchWrapper>
+      <DeckWrapper>
+        {Object.entries(deckNames).map(([id, deck]) => (
+          <DeckBox key={id}>
+            <Link href="/">
+              <StyledLink>{deck.deckName}</StyledLink>
+            </Link>
+          </DeckBox>
+        ))}
+        <DeckBox>
+          <form>
+            <input
+              type="text"
+              placeholder="Name of deck"
+              onChange={changeFunction}
+              value={value}
+            />
+          </form>
+          <button onClick={submitFunction} type="button">
+            Create deck
+          </button>
+        </DeckBox>
+        <Modal modalVisible={modalState}>
+          <ModalText>Deckname can not be empty.</ModalText>
+          <button type="button" onClick={dismissModal}>
+            close
+          </button>
+        </Modal>
```

Comment:
Wouldn't it be easier to disable the submit button when the input is an empty string?

Reply:
Fixed